pandas/core/groupby/groupby.py

```diff
@@ -2448,6 +2449,9 @@ def pre_processor(vals: ArrayLike) -> tuple[np.ndarray, np.dtype | None]:
             elif is_timedelta64_dtype(vals.dtype):
                 inference = np.dtype("timedelta64[ns]")
                 out = np.asarray(vals).astype(float)
+            elif isinstance(vals, ExtensionArray) and is_float_dtype(vals):
+                inference = np.dtype(np.float64)
```

Comment:
since we dont cast float32-> float64, I think we should cast Float32->float32 right?

Reply:
actually I think we cast float32 to float64. (When calling `_get_cythonized_result` we pass the below:
https://github.com/pandas-dev/pandas/blob/8814de9fbc7d53c6db201eab717fd3d6b607654b/pandas/core/groupby/groupby.py#L2474 
Below is the behaviour, I am getting the below on master.
```python
In [5]: import pandas as pd

In [6]: pd.__version__
Out[6]: '1.4.0.dev0+494.g5600a2fabb'

In [7]: df = pd.DataFrame({
   ...:     "x": [1, 1],
   ...:     "y": [0.2, np.nan]
   ...: }, dtype="float32")

In [8]: df.groupby("x")["y"].quantile(0.5)
Out[8]: 
x
1.0    0.2
Name: y, dtype: float64
```

Reply:
makes sense, thanks